connect/runtime/src/main/java/org/apache/kafka/connect/util/LoggingContext.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.util;
+
+import org.slf4j.MDC;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A utility for defining Mapped Diagnostic Context (MDC) for SLF4J logs.
+ *
+ * <p>Logging contexts may be nested. In this case, the nested context may override some or all of the parameters of the outer context.
+ * In these cases, both contexts may be active but the nested context takes priority. It is presumed that the nexted context will be
+ * stopped before the outer context.
+ *
+ */
+public final class LoggingContext implements AutoCloseable {
+
+    /**
+     * The parameter keys used by Connect.
+     */
+    public static final class Parameters {
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines the type of <i>connector</i>.
+         */
+        public static final String CONNECTOR_TYPE = "connector.type";
```

Comment:
Yeah, that should probably be changed. It is not "source" vs "sink", but rather a shortened name of the connector class. I'm actually not convinced this adds much value in the log compared to the connector name, and I've thought about removing it altogether to simplify things. 

The only reason I've kept it is because one could then easily filter the log to get all logs associated with a type of connector. Is that really useful?

Reply:
Changed with the latest commit.